app/src/main/java/org/thoughtcrime/securesms/SessionDialogBuilder.kt

```diff
@@ -149,22 +192,139 @@ class SessionDialogBuilder(val context: Context) {
 
 fun Context.showSessionDialog(build: SessionDialogBuilder.() -> Unit): AlertDialog =
     SessionDialogBuilder(this).apply { build() }.show()
-fun Context.showOpenUrlDialog(build: SessionDialogBuilder.() -> Unit): AlertDialog =
-    SessionDialogBuilder(this).apply {
-        title(R.string.urlOpen)
-        text(R.string.urlOpenBrowser)
-        build()
-    }.show()
 
-fun Context.showOpenUrlDialog(url: String): AlertDialog =
-    showOpenUrlDialog {
-        okButton { openUrl(url) }
-        cancelButton()
-    }
+public fun Context.copyURLToClipboard(url: String) {
+    val clipboard = getSystemService(Context.CLIPBOARD_SERVICE) as ClipboardManager
+    val clip = ClipData.newPlainText(url, url)
+    clipboard.setPrimaryClip(clip)
+}
+
+// Method to show a dialog used to open or copy a URL
+fun Context.showOpenUrlDialog(url: String, showCloseButton: Boolean = true): AlertDialog {
```

Comment:
Tested out this dialog and it doesn't look quite right. There's a scrollbar with no scrolling content, and the 'x' button isn't right (visually and in its location in relations to the title - see our composable preview in AlertDialog)
We have Composable dialogs now which could achieve the proper look easily.
It might be a bit harder to mingle in the xml setup but it's possible with the ComposeView.